internal/webhook/core/v1/pod_webhook.go

```diff
@@ -0,0 +1,188 @@
+		rayTLSGeneratorScript := getRayTLSGeneratorScriptInitContainer()
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, *rayTLSGeneratorScript)
+	}
+	return nil
+}
+
+// Add ray tls generator init-container into the pod when the pod has pod environment variable "RAY_USE_TLS" is 1.
+func getRayTLSGeneratorScriptInitContainer() *corev1.Container {
+
+	script := `
+echo "Generating Self Signed Certificate for Ray nodes"
+REFORMAT_IP=$(echo ${POD_IP} | sed 's+\.+\\.+g')
+JSONPATH={.data.$REFORMAT_IP}
+RAY_PEM_CONTENT=""
+RAY_CA_PEM_CONTENT=$(oc get secret ray-tls -n $POD_NAMESPACE -o jsonpath="{.data.ca\.crt}")
+TARGET_RAY_CA_PEM_FILE_PATH="/etc/ray/tls/ca.crt"
+TARGET_RAY_TLS_PEM_FILE_PATH="/etc/ray/tls/tls.pem"
+MAX_RETRIES=15
+RETRY_INTERVAL=2
+retries=0
+
+while [ $retries -lt $MAX_RETRIES ]; do
+  RAY_PEM_CONTENT=$(oc get secret ray-tls -n $POD_NAMESPACE -o jsonpath="$JSONPATH")
+  if [[ -n $RAY_PEM_CONTENT ]]; then
+    break
+  fi
+  retries=$((retries + 1))
+  echo "Cert file not generated yet. Retrying in ${RETRY_INTERVAL} seconds... ($retries/$MAX_RETRIES)"
+  sleep $RETRY_INTERVAL
+done
+if [ $retries -eq $MAX_RETRIES ]; then
+  echo "Error: Cert file not generated!"
+  exit 1
+fi
+
+echo $RAY_PEM_CONTENT|base64 -d > $TARGET_RAY_TLS_PEM_FILE_PATH
+echo $RAY_CA_PEM_CONTENT|base64 -d > $TARGET_RAY_CA_PEM_FILE_PATH
+
+if [ -f "$TARGET_RAY_TLS_PEM_FILE_PATH" ] && [ -f "$TARGET_RAY_CA_PEM_FILE_PATH" ] ; then
+  echo "Certificate files created successfully!"
+  exit 0
+else
+  echo "Error: Failed to create certificate files."
+  exit 1
+fi
+`
+	return &corev1.Container{
+		Name:  constants.RayTLSGeneratorInitContainerName,
+		Image: "registry.redhat.io/openshift4/ose-cli@sha256:25fef269ac6e7491cb8340119a9b473acbeb53bc6970ad029fdaae59c3d0ca61",
+		Command: []string{
+			"sh",
+			"-c",
+			script,
+		},
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_IP",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "status.podIP",
+					},
+				},
+			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "ray-tls",
+				MountPath: "/etc/ray/tls",
+			},
+		},
+	}
+}
+
+func needMutate(pod *corev1.Pod) bool {
+	return needToAddRayTLSGenerator(pod)
+}
+
+func needToAddRayTLSGenerator(pod *corev1.Pod) bool {
+	// Check if the pod already has ray tls generator init container.
+	for _, initContainer := range pod.Spec.InitContainers {
+		if initContainer.Name == constants.RayTLSGeneratorInitContainerName {
+			return false
+		}
+	}
+
+	// Check if RAY_USE_TLS is set to 1 in the main containers
+	for _, container := range pod.Spec.Containers {
+		if container.Name == "kserve-container" || container.Name == "worker-container" {
```

Comment:
isn't there a constant with the default container names?
if not, it might be a good idea to add.